src/devices/ServoMotor/ServoMotor.cs

```diff
@@ -0,0 +1,114 @@
+            _maximumPulseWidthMicroseconds = maximumPulseWidthMicroseconds;
+            _angleInMicroseconds = CalculateAngleToMicroseconds(maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds);
+        }
+
+        /// <summary>
+        ///  Initializes a new instance of the <see cref="ServoMotor"/> class.
+        /// </summary>
+        /// <param name="pwmChannel">The PWM channel used to control the servo motor.</param>
+        /// <param name="maximumAngle">The maximum angle the servo motor can move.</param>
+        /// <param name="minimumPulseWidthMicroseconds">The minimum pulse width, in microseconds, that represent an angle for 0 degrees.</param>
+        /// <param name="maximumPulseWidthMicroseconds">The maxnimum pulse width, in microseconds, that represent an angle for maximum angle.</param>
+        public ServoMotor(
+            PwmChannel pwmChannel,
+            int maximumAngle = 180,
+            int minimumPulseWidthMicroseconds = 1_000,
+            int maximumPulseWidthMicroseconds = 2_000)
+            : this(maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds)
+        {
+            _pwmChannel = pwmChannel;
+        }
+
+        private static double CalculateAngleToMicroseconds(int maximumAngle, int minimumPulseWidthMicroseconds, int maximumPulseWidthMicroseconds)
+        {
+            if (maximumAngle < 0 || maximumAngle > 360)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maximumAngle), maximumAngle, "Value must be between 0 and 360.");
+            }
+
+            if (minimumPulseWidthMicroseconds < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(minimumPulseWidthMicroseconds), minimumPulseWidthMicroseconds, "Value must not be negative.");
+            }
+
+            if (maximumPulseWidthMicroseconds < minimumPulseWidthMicroseconds)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maximumPulseWidthMicroseconds), maximumPulseWidthMicroseconds, $"Value must be greater than or equal to {minimumPulseWidthMicroseconds}.");
+            }
+
+            // slope = (y2 - y1) / (x2 - x1).  x1 is 0 in this case, so it is not subtracted.
+            double servoAngleSlope = (double)(maximumPulseWidthMicroseconds - minimumPulseWidthMicroseconds) / maximumAngle;
+            return servoAngleSlope;
+        }
+
+        /// <summary>
+        /// Starts the servo motor.
+        /// </summary>
+        public void Start() => _pwmChannel.Start();
+
+        /// <summary>
+        /// Stops the servo motor.
+        /// </summary>
+        public void Stop() => _pwmChannel.Stop();
+
+        /// <summary>
+        /// Writes an angle to the servo motor.
+        /// </summary>
+        /// <param name="angle">The angle to write to the servo motor.</param>
+        public void WriteAngle(int angle)
+        {
+            if (angle < 0 || angle > _maximumAngle)
```

Comment:
Same comment about casting to uint to make the call faster.

Reply:
Will update in next commit